### ISSUE TITLE
Translations of our CoC

### ISSUE THREAD

**Issue by ionas** [cakephp/code-of-conduct#2]
What about those?

In case there are none(?) I am willing to contribute the German one.

**Comment by dereuromark** [comment]
Why is that necessary?
Everyone working with the code must understand English anyway.
Also, once you open that Pandora's box there is always a lack of correctness towards the original source, as we can see from the documentation.

**Comment by ionas** [comment]
Difference: Changes will happen rather seldom.

Reason: Language is one big barrier even -- and while I still don't like "or _other_ un**professional** behavior" and it should be "_similar_" IMHO -- a CoC should be a binding document for all contributors and activities and thus should be easily and fully understandable even in its nuances, even by those speaking English barely or badly.

**Comment by phpnut** [comment]
@ionas I would not bother with translating it. The license, readme and contributing files are left untranslated, as are many other documents in the repos. Once we have a document that has no changes we can discuss it again but I see no need, unless we have a large group of non English speakers requesting it. No need to add additional work right?

**Comment by ceeram** [comment]
There already exist translations, here: https://github.com/CoralineAda/contributor_covenant/tree/master/version/1/3/0

A PR for german translation would best be made there in my opinion.

**Comment by beporter** [comment]
@ceeram: We did not adopt the Contributor Covenant so those translations are irrelevant to us. We do need to potentially generate our own now. 

I can see how you might be confused; @phpNut merged the Angular CoC, which bears little resemblance to the Contributor Covenant.

**Comment by ionas** [comment]
- It is still related to CC and parts of the translation could be at least used/looked at (will help).
- What about doing the translation as a joint effort with the angular team managing their CoC?

**Comment by beporter** [comment]
I do actually agree with @ionas that saying, "the license and the readme are English-only so why do we have to worry about the CoC?" is the lazy way out in this case.

For a document that has the express purpose of claiming that the project is open to all people and creeds to simultaneously say, "...as long as you speak English," is really the height of privilege talking. 

We are all fortunate enough to be able to discuss all of this in the defacto language of the industry, so it might be hard to think about someone who isn't even able to participate in these discussions about how much we "care" about them feeling welcomed to contribute. We aren't going to hear those voices because we are speaking English. But taking the time and effort to adjust our thinking about those underrepresented groups is supposedly **exactly** what we are doing here-- unless this is just a marketing move like @ionas originally feared.

**Comment by markstory** [comment]
I like the idea of translations, but worry that we'll run into the same issues we have with the docs where translations quickly fall out of sync with each other.

**Comment by ionas** [comment]
I think we should commit to a very slow adoption process for the CoC. After setup only one change a year. It is also a contract kind of thing and changing that often isn't good for the community IMHO, anyway.

**Suggestion A**: Setup a core set of languages to support and a change in the CoC will only be "official" by a combined PR (or set of PRs) with the updates. It should not be much but _for instance_ Spanish, simplified Chinese/Mandarin, US English, Russian, Japanese and French as most people can relate to any of those on the globe. Primary language (base of translation) will still be US English. We could also collaborate with Angular and those adopted Angular's CoC.

**Suggestion B**: Another way would be to postfix the code of conduct file names by versions followed up by language. As the basic spirit will probably not change too much _over time_, it should be a frame of reference for the most part.

**TLDR:**
Knowing the license is important if you do commercial business, to make sure the license does not counter your business requirements. For GPL the matter would be a bit more difficult (I favour especially the LGPL still, but MIT/BSD works, too).

In contrast a CoC should actually be read by anyone interested into actively working as part of the CakePHP community be it non-commercial or commercial. The whole list of objections (and even the dislikes on having a CoC) stem from this principle. Thus a CoC's meaning should REALLY be understood by all participants. Having a CoC available in English only is already the first step of non-inclusive practice and discredits the CoC document itself.

**Comment by beporter** [comment]
Yep, I still agree with @ionas here. 

@markstory: The problem of translations doesn't seem to have been an issue for the Contributor Covenant, which has now been translated into 9 other languages just since we've started this conversation. (EDIT: and [2 more in the works](https://github.com/CoralineAda/contributor_covenant/pulls?utf8=%E2%9C%93&q=is%3Aopen+translation+) as of this writing.) So now there actually is a strategic benefit to using it verbatim (it relieves the Cake community of having to maintain translations itself.) However, if the feeling is that we have enough community members available representing enough languages, then we should at least put a call out in IRC and on Twitter asking for help translating what we have. 

Another thing to keep in mind is that the Book and the API docs evolve rapidly--daily almost. That's a tough moving target to keep up with. By comparison, the hope is that changes to the CoC would be much more gradual and incremental: A new word or sentence every few months maybe? That just doesn't feel like as big a workload as what we're already managing. And again, either we're going to send the message that we **want** the community to be welcoming and inclusive (and we're willing to work for that goal even when it's "hard" for us), or we're going to send the message that we want to pay the ideas of inclusiveness and diversity lip service (as long as even that doesn't take too much work.)

Anyway, it shouldn't come as any surprise that I still favor the Contributor Covenant. It continues to gain traction and their community is clearly able to produce quality translations pretty rapidly. It seems like a wise (and humble) thing to do to acknowledge that they probably have done a better job than we could. Now that we have a CoC in place (one that's even originally based on the CC in fact!) it seems like a much smaller move to adopt it verbatim then it was when we were considering what to start with.
